doc/15-event-listener.md

```diff
@@ -0,0 +1,103 @@
+# Event Listening in PHP
+
+Implementation for event listening using custom events and listeners. The primary purpose is to allow custom logic to be executed at various points in the application lifecycle.
+
+## Table of Contents
+
+- [Listener Implementation](#listener-implementation)
+- [Provided Events](#provided-events)
+- [Creating and Registering Custom User Events](#creating-and-registering-custom-user-events)
```

Comment:
Original Message from @lyrixx 
> I would invert theses two chapters. Creating custom events is IMHO very rare.
>
> BTW, Do you have a good use case for it? I tend to will to remove this support

Not really in my use case, but in fact, given the low level of logic, is it really inconvenient to prevent someone from doing it when needed?

Reply:
I don't like adding feature without a use case. So let's remove this for now, until someone come with a use case. Less code to maintain is always better

Reply:
I'm not really sure it's a big effort to maintain it, given that it's managed by the EventDispatcher provided by Symfony.
But let's remove it for now :)